Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -494,52 +499,97 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
             {
                 if (cmdrPrefList.Count == 0)
                 {
-                    if (prefList.Count == 0)
+
+                    if (medPrefList.Count == 0)
                     {
-                        if (everyone.Count == 0)
+                        if (prefList.Count == 0)
                         {
-                            Logger.InfoS("preset", "Insufficient ready players to fill up with nukeops, stopping the selection");
-                            break;
+                            if (everyone.Count == 0)
+                            {
+                                Logger.InfoS("preset", "Insufficient ready players to fill up with nukeops, stopping the selection");
+                                break;
+                            }
+                            nukeOp = _random.PickAndTake(everyone);
+                            Logger.InfoS("preset", "Insufficient preferred nukeop commanders, agents or nukies, picking at random.");
+                        }
+                        else
+                        {
+                            nukeOp = _random.PickAndTake(prefList);
+                            everyone.Remove(nukeOp);
+                            Logger.InfoS("preset", "Insufficient preferred nukeop commander or agents, picking at random from regular op list.");
                         }
-                        nukeOp = _random.PickAndTake(everyone);
-                        Logger.InfoS("preset", "Insufficient preferred nukeop commanders or nukies, picking at random.");
                     }
                     else
                     {
-                        nukeOp = _random.PickAndTake(prefList);
-                        everyone.Remove(nukeOp);
-                        Logger.InfoS("preset", "Insufficient preferred nukeop commanders, picking at random from regular op list.");
+                    nukeOp = _random.PickAndTake(medPrefList);
+                    everyone.Remove(nukeOp);
+                    prefList.Remove(nukeOp);
+                    Logger.InfoS("preset", "Insufficient preferred nukeop commanders, picking an agent")
                     }
+
                 }
                 else
                 {
                     nukeOp = _random.PickAndTake(cmdrPrefList);
                     everyone.Remove(nukeOp);
                     prefList.Remove(nukeOp);
+                    medPrefList.Remove(nukeOp);
                     Logger.InfoS("preset", "Selected a preferred nukeop commander.");
                 }
             }
             else
             {
-                if (prefList.Count == 0)
+                if (i == 1)
```

Comment:
combine this with the else `else if (i == 1)`

Reply:
what? how?

Reply:
```
if (x)
{
}
else
{
    if (y)
    {
    }
    else
    {
    }
}
```

is the same as

```
if (x)
{
}
else if (y)
{
}
else
{
}
```

Reply:
like in commit c7c385a ?